### PR TITLE
Style using the Foundation framework

### PR DESCRIPTION
**Pull request by katherine-boost**
This makes the amount of custom CSS much smaller.

Also:
• Pulled out list item (i.e., actual grocery item which is part of a shopping list) into a partial. This was just for readability.

Addresses feedback in #3:
* Generally just use Foundation
* Use `rem` rather than `em` or `px`
    * Exceptions: `border-radius` (I think `px` makes sense for that), and the Foundation breakpoints which are in `em`.

I did have some problems with Foundation - the `XY Grid` was not working whatsoever; neither could I use the new breakpoint syntax (e.g. `@include breakpoint(medium only)`).  
Per its repo, the Rails gem appears to be up to date, so it's pretty puzzling.

### Mobile views

shopping-list_homepage_mob
shopping-list_list_mob

### Desktop views

shopping-list_homepage_dt
shopping-list_list_dt

Okay it's still not very beautiful, but at least it's in Foundation now!

### CODE REVIEW

File: app/assets/stylesheets/application.css.scss [hunk 1]
```diff
@@ -16,76 +16,77 @@
 
  */
 @import 'foundation-icons';
+
+/* CUSTOM VARIABLES */
+
 $boost-red: #B72039;
```
Comment (tim-boost) [inline app/assets/stylesheets/application.css.scss]: Could be worth making a _variables.scss or _colours.scss file to store colours and then import in that file above

File: app/assets/stylesheets/application.css.scss [hunk 2]
```diff
@@ -16,76 +16,77 @@
...
 
+/* FOUNDATION BREAKPOINTS */
+
+$medium-start: 40.063em;
```
Comment (tim-boost) [inline app/assets/stylesheets/application.css.scss]: Could replace these with the ones from the foundation overrides file at some point

File: app/assets/stylesheets/application.css.scss [hunk 3]
```diff
@@ -16,76 +16,77 @@
...
+$large-start: 63.9375em;
+
+/* BASICS */
+
 body {
+  background-color: $boost-red;
   color: white;
```
Comment (tim-boost) [inline app/assets/stylesheets/application.css.scss]: Could make this a variable

Comment (tim-boost) [reply app/assets/stylesheets/application.css.scss]: Or use $primary-color

File: app/views/lists/show.html.erb [hunk 4]
```diff
@@ -44,16 +27,23 @@
...
+      <%= form.text_field :name, placeholder: :name %>
+    </div>
+    <div class="columns small-3 large-2">
+      <%= form.number_field :quantity, value: 1 %>
+    </div>
+    <div class="columns small-6 large-3">
+      <%= form.select(:category_id, options_for_select(@categories.map { |c| [c.name, c.id] })) %>
+    </div>
+    <div class="columns small-6 large-3">
+      <%= form.submit "Add item", class: "button small" %>
```
Comment (tim-boost) [inline app/views/lists/show.html.erb]: Better rails style to use single quotes here

File: app/views/lists/index.html.erb [hunk 5]
```diff
@@ -16,7 +20,7 @@
   <% end %>
 </div>
 
-<%= link_to lists_path, method: :post, class: "nav-link" do %>
+<%= link_to lists_path, method: :post, class: "button" do %>
```
Comment (tim-boost) [inline app/views/lists/index.html.erb]: Could use single quotes here

**Review by tim-boost — APPROVED** [review]

File: app/assets/stylesheets/application.css.scss (hunk 2 above)
Comment (katherine-boost) [reply app/assets/stylesheets/application.css.scss]: Done in next commit :)

File: app/assets/stylesheets/application.css.scss (hunk 3 above)
Comment (katherine-boost) [reply app/assets/stylesheets/application.css.scss]: do you mean make `background-color: $boost-red;` a variable in itself?

**Review by tim-boost — APPROVED** [review]

File: app/assets/stylesheets/application.css.scss (hunk 1 above)
Comment (katherine-boost) [reply app/assets/stylesheets/application.css.scss]: Done in 82a524a

File: app/views/lists/show.html.erb (hunk 4 above)
Comment (katherine-boost) [reply app/views/lists/show.html.erb]: Done in 82a524a (+ other instances)

File: app/views/lists/index.html.erb (hunk 5 above)
Comment (katherine-boost) [reply app/views/lists/index.html.erb]: Done in 82a524a

**Pull request merged**